### PR TITLE
nix: allow entering slim shell with packages compiled from local dune

### PR DESCRIPTION
**Pull request by anmonteiro**
- Removes our usage of opam-nix
- Preserves a package `nix build .#dune` or `nix develop .#scope` which uses the dune package from the repository to build the OCaml nixpkgs scope

### CODE REVIEW

**Comment by Alizter** [comment]
i'll try this ASAP.

File: flake.nix [hunk 1]
```diff
@@ -1,20 +1,13 @@
 {
   inputs = {
-    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+    # Dune 3.8 (https://github.com/NixOS/nixpkgs/pull/233595), required to
+    # compile Melange
+    nixpkgs.url = "github:nixos/nixpkgs/71317efbb568a58cc4168bda174662bb6fe2918f";
```
Comment (Alizter) [inline flake.nix:5]: Should be in unstable now.

**Comment by Alizter** [comment]
Works great! I'm happy to finally move away from opam-nix :)

**Comment by rgrinberg** [comment]
I think this is missing core_bench. So it's not possible to run the micro benchmarks. You can verify with `make check`.

**Comment by anmonteiro** [comment]
Added all the previous dependencies in the default shell, should have everything available now, including `core_bench` and `mdx`.

**Review by rgrinberg — APPROVED** [review]

**Review by Alizter — APPROVED** [review]

**Pull request merged**